### PR TITLE
feat: added /logs endpoint

### PR DESCRIPTION
**Pull request by adityathebe**
Resolves https://github.com/flanksource/canary-checker/issues/569

### CODE REVIEW

File: cmd/server.go [hunk 1]
```diff
@@ -105,7 +113,9 @@ var Serve = &cobra.Command{
 		forward(e, "/config", configDb)
 		forward(e, "/canary", api.CanaryCheckerPath)
 		forward(e, "/kratos", kratosAPI)
-		forward(e, "/apm", api.ApmHubPath)
+		forward(e, "/apm", api.ApmHubPath) // Deprecated
+
+		forwardWithMiddlewares(e, "/logs", api.ApmHubPath, logSearchMiddleware)
```
Comment (moshloop) [inline cmd/server.go:118]: I don't think this a middleware - those are normally generic -

File: cmd/server.go [hunk 2]
```diff
@@ -114,27 +124,139 @@ var Serve = &cobra.Command{
...
+			return fmt.Errorf("failed to parse form: %w", err)
+		}
+
+		componentID, ok := form["id"].(string)
+		if !ok {
+			return next(c) // No component ID, so no need to modify the request
+		}
+
+		selectorName, _ := form["name"].(string)
+		componentID = "01872d9f-29c6-3626-2da5-9851502cd6ba" // Test
```
Comment (moshloop) [inline cmd/server.go:154]: Need to fix

File: cmd/server.go (hunk 1 above)
Comment (adityathebe) [reply cmd/server.go]: modified to use a http handler instead

File: logs/handler.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+			Message: "failed to get component by the given ID",
+		})
+	}
+
+	selectorName, _ := form["name"].(string)
+	logSelector := getLabelsFromLogSelectors(component.LogSelectors, selectorName)
+	if logSelector == nil {
+		return c.JSON(http.StatusBadRequest, api.HTTPError{
+			Error:   "log selector was not found",
+			Message: fmt.Sprintf("log selector with the name '%s' was not found.", selectorName),
```
Comment (moshloop) [inline logs/handler.go:50]: Add in the valid names into the error message

File: logs/handler.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+		})
+	}
+
+	templater := ktemplate.StructTemplater{
```
Comment (moshloop) [inline logs/handler.go:54]: switch to commons/template/StructTemplatar

File: logs/handler.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+		return &selectors[0]
+	}
+
+	for _, selector := range selectors {
+		if selector.Name == name {
+			return &selector
+		}
+	}
+
+	logger.Debugf("could not find log selector with name %s", name)
```
Comment (moshloop) [inline logs/handler.go:126]: Don't log this

File: logs/handler.go [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+	component, err := duty.GetComponent(req.Context(), db.Gorm, componentID)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, api.HTTPError{
+			Error:   err.Error(),
+			Message: "failed to get component by the given ID",
+		})
+	}
+
+	selectorName, _ := form["name"].(string)
```
Comment (moshloop) [inline logs/handler.go:45]: Make the name optional - if none specified use the first log selector.

Comment (adityathebe) [reply logs/handler.go:45]: @moshloop it's already like that.

**Pull request merged**